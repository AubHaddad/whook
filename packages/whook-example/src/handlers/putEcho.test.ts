import initPutEcho, { definition } from './putEcho';
import YError from 'yerror';
import { OpenAPIV3 } from 'openapi-types';

describe('putEcho', () => {
  const log = jest.fn();

  it('should work', async () => {
    const putEcho = await initPutEcho({
      log,
    });
    const response = await putEcho({
      body: (definition.operation.requestBody as OpenAPIV3.RequestBodyObject)
        .content['application/json'].example,
    });

    expect({
      response,
    }).toMatchInlineSnapshot(`
      Object {
        "response": Object {
          "body": Object {
            "echo": "Repeat this!",
          },
          "status": 200,
        },
      }
    `);
    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });

  it('should fail when crossing the red line ;)', async () => {
    const putEcho = await initPutEcho({
      log,
    });

    try {
      await putEcho({
        body: { echo: 'Big up to Lord Voldemort!' },
      });
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect({
        errorCode: err.code,
        errorParams: err.params,
      }).toMatchInlineSnapshot(`
        Object {
          "errorCode": "E_MUST_NOT_BE_NAMED",
          "errorParams": Array [
            "Big up to Lord Voldemort!",
          ],
        }
      `);
      expect({
        logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      }).toMatchSnapshot();
    }
  });
});
