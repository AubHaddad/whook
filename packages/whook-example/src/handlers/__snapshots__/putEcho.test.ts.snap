// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`putEcho should fail when crossing the red line ;) 2`] = `
Object {
  "logCalls": Array [
    Array [
      "warning",
      "ðŸ“¢ - Echoing \\"Repeat this!\\"",
    ],
  ],
}
`;

exports[`putEcho should work 2`] = `
Object {
  "logCalls": Array [
    Array [
      "warning",
      "ðŸ“¢ - Echoing \\"Repeat this!\\"",
    ],
  ],
}
`;
